pkgs/development/libraries/jemalloc/rust.nix

```diff
@@ -0,0 +1,22 @@
+{ stdenv, jemalloc, writeText }:
+
+jemalloc.overrideAttrs (oldAttrs: {
+  configureFlags = oldAttrs.configureFlags ++ [
+    "--with-jemalloc-prefix=_rjem_"
+    "--with-private-namespace=_rjem_"
+  ];
+
+  postFixup = ''
+    touch $out/rust-jemalloc
+  '';
+
+  setupHook = writeText "setup-hook.sh" ''
+    addRustJemalloc () {
+      if test -f "$1/rust-jemalloc"; then
+        export JEMALLOC_OVERRIDE="$1/lib/libjemalloc${stdenv.hostPlatform.extensions.library}"
```

Comment:
I think I'm probably missing something, but would `export JEMALLOC_SYS_WITH_LG_PAGE=16` not work here?

Reply:
If you look at the options passed `pkgs/development/libraries/jemalloc/default.nix`, they are applying more fixes specifically for arm 32-bit, x86 darwin, and enabling up to 64K pagesize.
We could pass all of these options, but  then it gets more complicated to keep them in sync, compared to just one JEMALLOC_OVERRIDE.